olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp

```diff
@@ -326,6 +329,109 @@ void DataRepository::GetData(
                   *cache_);
 }
 
+BlobApi::DataResponse DataRepository::GetBlobData(
+    const client::HRN& catalog, const std::string& layer,
+    const DataRequest& data_request,
+    client::CancellationContext cancellation_context,
+    client::OlpClientSettings settings) {
+  using namespace client;
+
+  auto fetch_option = data_request.GetFetchOption();
+  const auto& data_handle = data_request.GetDataHandle();
+
+  if (!data_handle) {
+    auto key = data_request.CreateKey();
+    OLP_SDK_LOG_INFO_F(kLogTag, "clear '%s' cache", key.c_str());
+    return ApiError(ErrorCode::PreconditionFailed, "Data handle is missing");
+  }
+
+  repository::DataCacheRepository repository(catalog, settings.cache);
+
+  if (fetch_option != OnlineOnly) {
+    auto cached_data = repository.Get(layer, data_handle.value());
+    if (cached_data) {
+      auto key = data_request.CreateKey();
+      OLP_SDK_LOG_INFO_F(kLogTag, "cache data '%s' found!", key.c_str());
```

Comment:
```suggestion
      OLP_SDK_LOG_INFO_F(kLogTag, "cache data '%s' found!", data_request.CreateKey().c_str());
```
We shouldn't create a key if logging is disabled.